File: atomic/src/main/scala/resonantinduction/atomic/items/ItemCell.java
package resonantinduction.atomic.items;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.client.renderer.texture.IconRegister;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;
import resonant.api.IReactor;
import resonant.lib.prefab.item.ItemTooltip;
import resonant.lib.utility.LanguageUtility;
import resonantinduction.atomic.Atomic;

public class ItemCell extends ItemTooltip
{
    public ItemCell(int itemID)
    {
        super(itemID);
        setContainerItem(Atomic.itemCell);
    }

    @Override
    public boolean shouldPassSneakingClickToBlock(World world, int x, int y, int z)
    {
        return world.getBlockTileEntity(x, y, z) instanceof IReactor;
    }
    
    @SideOnly(Side.CLIENT)
    @Override
    public void registerIcons(IconRegister iconRegister)
    {
        this.itemIcon = iconRegister.registerIcon(this.getUnlocalizedName().replace("item.", ""));
    }

    @Override
    public String getUnlocalizedName(ItemStack itemstack)
    {
        String localized = LanguageUtility.getLocal(getUnlocalizedName() + "." + itemstack.getItemDamage() + ".name");
        if (localized != null && !localized.isEmpty())
        {
            return getUnlocalizedName() + "." + itemstack.getItemDamage();
        }
        
        return getUnlocalizedName();
    }
}

File: src/main/scala/edx/quantum/items/ItemCell.scala
package edx.quantum.items

import edx.core.{EDXCreativeTab, Reference}
import edx.quantum.QuantumContent
import net.minecraft.item.ItemStack
import resonantengine.lib.prefab.item.ItemTooltip
import resonantengine.lib.utility.LanguageUtility

class ItemCell extends ItemTooltip
{
  //Constructor
  setContainerItem(QuantumContent.itemCell)

  def this(name: String)
  {
    this()
    if (!name.equalsIgnoreCase("cellEmpty")) this.setContainerItem(QuantumContent.itemCell)
    this.setUnlocalizedName(Reference.prefix + name)
    this.setTextureName(Reference.prefix + name)
    setCreativeTab(EDXCreativeTab)
  }

  override def getUnlocalizedName(itemstack: ItemStack): String =
  {
    val localized: String = LanguageUtility.getLocal(getUnlocalizedName() + "." + itemstack.getItemDamage + ".name")
    if (localized != null && !localized.isEmpty)
    {
      return getUnlocalizedName() + "." + itemstack.getItemDamage
    }
    return getUnlocalizedName()
  }
}